langlib/langlib-test/src/test/resources/test-src/xmllib_test.bal

```diff
@@ -382,6 +397,53 @@ function testElementChildrenNS() {
     assert(toNoNs.toString(), "<to>Irshad</to><to>Irshad</to>");
 }
 
+function testXMLIteratorInvocation() {
+    xml a = xml `<!--first-->`;
+    xml<'xml:Comment> seq1 = <xml<'xml:Comment>> a.concat(xml `<!--second-->`);
+
+    object {
+        public isolated function next() returns record {| 'xml:Comment value; |}?;
+    } iter1 = seq1.iterator();
+
+    assert((iter1.next()).toString(), "{\"value\":`<!--first-->`}");
```

Comment:
We should ideally test iterating the entire sequence. Including `()` being returned after everything has been iterated.